### PR TITLE
Fixed deleteProject handler

### LINKED ISSUE #3023: `deleteProject` handler doesn't wait for S3 deletion to complete
### Increasing Access

It seems like we could end up with files left behind on S3, which use up resources.

### Feature enhancement details

The `deleteProject` handler fires off an asynchronous function `deleteObjectsFromS3` (via `deleteFilesFromS3`) but it does not wait for the S3 deletion to continue before sending an API response.

I'm not sure what actually happens here -- do those files continue to get deleted in the background or not?  I'm concerned that we could wind up with "leftover" files on S3 that are no longer linked to any project.

There is no waiting here:
https://github.com/processing/p5.js-web-editor/blob/6cac275429c3911d87afdcdbbdbf185092aeac29/server/controllers/project.controller/deleteProject.js#L61-L72

I noticed this thanks to a warning in my IDE 
image

### PR DESCRIPTION
**Pull request by letscodedanish**
Fixes #3023 

Changes:
In this modification, I added a return statement before deleteObjectsFromS3 to ensure the function returns the Promise returned by deleteObjectsFromS3. Then, I modified the handleProjectDeletion function to wait for the delete operation to complete using .then() and .catch() blocks to handle any errors during the S3 deletion process.

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] has no test errors (`npm run test`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: server/controllers/project.controller/deleteProject.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const ProjectDeletionError = createApplicationErrorClass(
 );
 
 function deleteFilesFromS3(files) {
-  deleteObjectsFromS3(
+  // Return the Promise returned by deleteObjectsFromS3
+  return deleteObjectsFromS3(
```
Comment (lindapaiste) [inline server/controllers/project.controller/deleteProject.js:12]: Perfect 👍

File: server/controllers/project.controller/deleteProject.js [hunk 2]
```diff
@@ -59,16 +60,25 @@ export default function deleteProject(req, res) {
...
-      if (removeProjectError) {
-        sendProjectNotFound();
-        return;
-      }
+    // Wait for the delete operation to complete before proceeding
+    deleteFilesFromS3(project.files)
+      .then(() => {
+        project.remove((removeProjectError) => {
+          if (removeProjectError) {
+            sendProjectNotFound();
```
Comment (lindapaiste) [inline server/controllers/project.controller/deleteProject.js:68]: Just want to link this line to issue #3024 to be fixed there -- "not found" isn't the right error.  It should be a 500.

File: server/controllers/project.controller/deleteProject.js [hunk 3]
```diff
@@ -59,16 +60,25 @@ export default function deleteProject(req, res) {
...
+          res.status(200).end();
+        });
+      })
+      .catch((error) => {
+        sendFailure(
+          new ProjectDeletionError('Error deleting files from S3', {
+            code: 500
+          })
+        );
+      });
```
Comment (lindapaiste) [inline server/controllers/project.controller/deleteProject.js:81]: Great error handling!

**Review by lindapaiste — COMMENTED** [review]
This looks good but I am extremely confused by why the Git diff is showing all the changes from the SASS PR.  I think I can fix it for you (by merging the current branch into your branch) but not right now.

**Comment by letscodedanish** [comment]
> This looks good but I am extremely confused by why the Git diff is showing all the changes from the SASS PR. I think I can fix it for you (by merging the current branch into your branch) but not right now.

Thank you @lindapaiste  for looking into it. I appreciate your effort to resolve the confusion. Please take your time, and let me know if you need any further assistance from my end.

**Comment by letscodedanish** [comment]
> This looks good but I am extremely confused by why the Git diff is showing all the changes from the SASS PR. I think I can fix it for you (by merging the current branch into your branch) but not right now.

Thank you @lindapaiste  for bringing this to my attention. It seems there might be a discrepancy in the Git diff, possibly showing changes from the SASS PR that aren't directly related to our current task, I think there could be mainly three reasons:
1. It's possible that changes from other branches, such as the SASS PR, may be inadvertently included during the branch creation or merging process.
2. The Git history might have become tangled or unclear, leading to unexpected changes being displayed in the diff.
3. If multiple components or features are being developed concurrently in the repository, it's easier for changes to overlap or conflict in the diff view.

To address this, we may need to review the Git history and ensure that proper branch management practices are followed. There might be some unnecessary changes or merges that need to be cleaned up. Your help in resolving this issue is greatly appreciated.

**Comment by letscodedanish** [comment]
> This looks good but I am extremely confused by why the Git diff is showing all the changes from the SASS PR. I think I can fix it for you (by merging the current branch into your branch) but not right now.

have a look on this @lindapaiste , Please take your time, and let me know if you need any further assistance from my end.

**Comment by raclim** [comment]
Thanks for your work on this! Due to the amount of time that's passed I'm going to close this PR for now. I'm sorry that we couldn't get this in, but please feel free to reopen a new one or check out our other issues!

**Pull request closed without merging**